pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -518,3 +505,68 @@ func (ch *ClickHouseExportProcess) batchCommitAll() (int, error) {
 
 	return currSize, nil
 }
+
+func CheckConnection(input ClickHouseInput) (string, *sql.DB, error) {
+	dsn, err := input.getDataSourceName()
+	if err != nil {
+		klog.ErrorS(err, "Error parsing ClickHouse DSN")
```

Comment:
in general, you should not both log an error and also return it, as it causes errors to be eventually logged twice.
in this case, you just want to return the error, and possibly wrap it to add more context

applies to the rest of this function as well

Reply:
Fixed. Thanks.